analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -737,6 +737,39 @@ void RefParams(int? iParam)
     }
 }
 
+class InParams
+{
+    void InParamOfALocalFunction(int? iParam)
+    {
+        iParam = null;
+        LocalFunctionWithReadInParam(iParam);
+        _ = iParam.Value;       // Noncompliant, empty
+
+        iParam = 42;
+        LocalFunctionWithReadInParam(iParam);
+        _ = iParam.Value;       // Compliant, non-empty
+
+        static void LocalFunctionWithReadInParam(in int? i) { }
+    }
+
+    void InParamOfTheMethodItself(in int? iParam)
+    {
+        _ = iParam.Value;       // Compliant, unknown
+        iParam = null;          // Error, CS8331
+        _ = iParam.Value;       // Compliant, iParam is "in" and its value can't be modified
+    }
+
+    void InParamOfAnotherMethod(in int? iParam)
+    {
+        _ = iParam.Value;       // Compliant, unknown
+
+        MethodWithReadInParam(iParam);
+        _ = iParam.Value;       // Compliant, iParam is "in" and its value can't be modified
+    }
```

Comment:
We need this test case where `iParam` is a local known to be null. Then the call is done and the value is passed:
```suggestion
    void InParamOfAnotherMethod(in int? iParam)
    {
        _ = iParam.Value;       // Compliant, unknown

        MethodWithReadInParam(iParam);
        _ = iParam.Value;       // Compliant, iParam is "in" and its value can't be modified
    }
    
    void InParamToAnotherMethod()
    {
        int? arg = null;
        MethodWithReadInParam(arg);
        _ = arg.Value;       // Noncompliant, arg is "in" and its value must still be null here
    }    
```

Reply:
Added, I called it `LocalAsInParamToAnotherMethod`, to make clear that the `in` param is `in` for the other method, and not for the method which is being executed.